fp_filter/workflow.cwl

```diff
@@ -31,7 +40,33 @@ steps:
             reference: reference
             bam: cram_to_bam/bam
             vcf: vcf
+            variant_caller: variant_caller
+            output_vcf_basename: 
+                valueFrom: $(inputs.variant_caller)_unfiltered
+        out:
+            [filtered_vcf]
+    fp_bgzip:
+        run: ../detect_variants/bgzip.cwl
+        in:
+            file: fp_filter/filtered_vcf
+        out:
+            [bgzipped_file]
+    fp_index:
+        run: ../detect_variants/index.cwl
+        in:
+            vcf: fp_bgzip/bgzipped_file
+        out:
+            [indexed_vcf]
+    hard_filter:
+        run: ../detect_variants/select_variants.cwl
+        in:
+            reference: reference
+            vcf: fp_index/indexed_vcf
+            exclude_filtered: 
+                default: true
+            variant_caller: variant_caller
+            output_vcf_basename: 
+                valueFrom: $(inputs.variant_caller)_filtered
```

Comment:
Using the `StepInputExpressionRequirement`, I believe the dependency on the other input could be removed in favor of something like:

```yaml
output_vcf_basename:
  source: variant_caller
  valueFrom: "$(self)_filtered"
```

Per [this section of the docs](http://www.commonwl.org/v1.0/Workflow.html#WorkflowStepInput), `self` must refer to the value of the source (or `null` if there is no source).